### ISSUE TITLE
IDIDP004 should not warn when returning wrapper that disposes.

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#200]
```cs
namespace N
{
    using System.IO;

    public static class C
    {
        public StreamReader M1() => File.OpenRead(string.Empty)
                                        .M2();

        private static StreamReader M2(this Stream stream) => new StreamReader(stream);
    }
}
```

This is a regression.
